src/Theming/ThemeSupport.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * ThemeSupport class file.
+ *
+ * @package WooCommerce\Theming
+ */
+
+namespace Automattic\WooCommerce\Theming;
+
+use Automattic\WooCommerce\Utils\ArrayUtils;
+
+/**
+ * Provides methods for theme support.
+ * Get an instance via WooCommerce::theme_support().
+ *
+ * @package Automattic\WooCommerce\Theming
+ */
+class ThemeSupport {
+
+	const DEFAULTS_KEY = '_defaults';
+
+	/**
+	 * Holds the only instance of the class.
+	 *
+	 * @var ThemeSupport The only instance of the class.
+	 */
+	private static $instance;
+
+	/**
+	 * Returns the only existing instance of the class.
+	 *
+	 * @return ThemeSupport The only existing instance of the class.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Adds theme support options for the current theme.
+	 *
+	 * @param array $options The options to be added.
+	 */
+	public function add_theme_support_options( $options ) {
```

Comment:
How would you feel about removing the `theme_support` from every method in this class? It seems redundant that the `ThemeSupport` class has a `add_theme_support_options` method instead of just an `add_options` method.

Reply:
Yep, that makes sense. I read too much non-object-oriented code already and it's spoiling my naming abilities. 😅